tests/components/evil_genius_labs/test_config_flow.py

```diff
@@ -0,0 +1,60 @@
+"""Test the Evil Genius Labs config flow."""
+from unittest.mock import patch
+
+from homeassistant import config_entries
+from homeassistant.components.evil_genius_labs.config_flow import CannotConnect
+from homeassistant.components.evil_genius_labs.const import DOMAIN
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import RESULT_TYPE_CREATE_ENTRY, RESULT_TYPE_FORM
+
+
+async def test_form(hass: HomeAssistant, data_fixture) -> None:
+    """Test we get the form."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["errors"] is None
+
+    with patch(
+        "pyevilgenius.EvilGeniusDevice.get_data",
+        return_value=data_fixture,
+    ), patch(
+        "homeassistant.components.evil_genius_labs.async_setup_entry",
+        return_value=True,
+    ) as mock_setup_entry:
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                "host": "1.1.1.1",
+            },
+        )
+        await hass.async_block_till_done()
+
+    assert result2["type"] == RESULT_TYPE_CREATE_ENTRY
+    assert result2["title"] == "Fibonacci256-23D4"
+    assert result2["data"] == {
+        "host": "1.1.1.1",
+    }
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_form_cannot_connect(hass: HomeAssistant) -> None:
+    """Test we handle cannot connect error."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch(
+        "pyevilgenius.EvilGeniusDevice.get_data",
+        side_effect=CannotConnect,
```

Comment:
Don't patch with our error. Set the error that the library will use, ie aiohttp.ClientError.